src/scent_map.cpp

```diff
@@ -76,28 +76,26 @@ void scent_map::shift( const int sm_shift_x, const int sm_shift_y )
     scent_array<int> new_scent;
     for( size_t x = 0; x < SEEX * MAPSIZE; ++x ) {
         for( size_t y = 0; y < SEEY * MAPSIZE; ++y ) {
-            // operator() does bound checking and returns 0 upon invalid coordinates
-            new_scent[x][y] = operator()( x + sm_shift_x, y + sm_shift_y );
+            // get does bound checking and returns 0 upon invalid coordinates
+            new_scent[x][y] = get( x + sm_shift_x, y + sm_shift_y );
         }
     }
     grscent = new_scent;
 }
 
-int scent_map::operator()( const size_t x, const size_t y ) const
+int scent_map::get( const size_t x, const size_t y ) const
 {
     if( inbounds( x, y ) ) {
         return grscent[x][y];
     }
     return 0;
 }
 
-int &scent_map::operator()( const size_t x, const size_t y )
+void scent_map::set( const size_t x, const size_t y, int value )
 {
     if( inbounds( x, y ) ) {
-        return grscent[x][y];
+        grscent[x][y] = value;
     }
-    null_scent = 0;
```

Comment:
Seems you can remove the `null_scent` from the class definition as it's not used anymore.